### ISSUE TITLE
[Idea] DLC Exclusive Items

### ISSUE THREAD

**Issue by TheNasbit** [Razzmatazzz/RemnantSaveGuardian#239]
Mark Items as DLC Exclusive (and what DLC they are from) and an option to filter DLC Exclusive items.

**Comment by AndrewSav** [comment]
Is there a list anywhere which show which are which? IMO some items added in a DLC are note DLC exclusive, so might be challenging to get this info

**Comment by TheNasbit** [comment]
Have no idea yet. It was an thought when i saw an banner in wiki. ![Screenshot_20240501-160819.jpg](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/52046382/1398dc7b-4ab9-4635-8a08-e256022913e9)

**Comment by razeedazee** [comment]
I do not own the DLC's and will be able to test this.

What do you need?

**Comment by TheNasbit** [comment]
> I do not own the DLC's and will be able to test this.
> 
> What do you need?

Information if any DLC items that are random drops on other worlds will generated for you. So can you find any DLC Item when you're playing on your world (not joining others in multiplayer) 

Taking N'Erud as example:
Finding any of the following:
- [Legacy Protocol](https://remnant2.wiki.fextralife.com/Legacy+Protocol) (TAK)
- [Power Complex](https://remnant2.wiki.fextralife.com/Power+Complex) (TAK)
- [A'taerii Booster](https://remnant2.wiki.fextralife.com/A'taerii+Booster) (TAK)
- [Zero Divide](https://remnant2.wiki.fextralife.com/Zero+Divide) (TFK)
- [Closed Loop](https://remnant2.wiki.fextralife.com/Closed+Loop) (TFK)
- [Ring of Ordnance](https://remnant2.wiki.fextralife.com/Ring+of+Ordnance) (TFK)

And one example of Losomn:
- [Magnifying Glass](https://remnant2.wiki.fextralife.com/Magnifying+Glass) (TFK)

**Comment by AndrewSav** [comment]
Also would be good to confirm if you can get any of those in multiplayer

**Comment by razeedazee** [comment]
![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/84c4adb0-2e15-40d9-a589-0d7fc7c843d5)


Legacy Protocol

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/54fbf024-f0be-4024-a39d-0dab12a50e35)

![2024-05-02_09-18-05](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/64bb2c4b-ef12-411b-804a-c7eb6b3a5453)

**Comment by razeedazee** [comment]
Power complex

![2024-05-02_10-05-22](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/e3275e8e-d6d2-42de-b7a1-66575c2cc505)

**Comment by TheNasbit** [comment]
Whats about [Magnifying Glass](https://remnant2.wiki.fextralife.com/Magnifying+Glass) on Losomn, because there BOTH wikis state that its exclusive to the DLC.

**Comment by razeedazee** [comment]
I am not seeing any TFK items in RSG, just the TAK stuff, 

[Legacy Protocol](https://remnant2.wiki.fextralife.com/Legacy+Protocol) (TAK) YES
[Power Complex](https://remnant2.wiki.fextralife.com/Power+Complex) (TAK) YES
[A'taerii Booster](https://remnant2.wiki.fextralife.com/A'taerii+Booster) (TAK) YES
[Zero Divide](https://remnant2.wiki.fextralife.com/Zero+Divide) (TFK) NO
[Closed Loop](https://remnant2.wiki.fextralife.com/Closed+Loop) (TFK) YES
[Ring of Ordnance](https://remnant2.wiki.fextralife.com/Ring+of+Ordnance) (TFK) NO
[Magnifying Glass](https://remnant2.wiki.fextralife.com/Magnifying+Glass) (TFK) NO

Would i see it in RSG if it was there? I have rerolled adventure mode a good few times and not seen the Magnifying Glass

**Comment by TheNasbit** [comment]
> Would i see it in RSG if it was there? I have rerolled adventure mode a good few times and not seen the Magnifying Glass

For DLC2 - sadly not yet - Not without the new update. After the update, yes.

**Comment by TheNasbit** [comment]
But if I see correctly: Random DLC Item Drops are generated even without the DLC.
You just get locked out from the DLC-Story-Lines? So would still a good idea to mark/filter DLC items in missing Item Tab. So people know what they can obtain with the DLCs they own.

**Comment by razeedazee** [comment]
I was doing targeted runs for the items shown in RSG, I'll have to do a full tour to see if the TFK drops are there. I'll do that in a bit.

**Comment by razeedazee** [comment]
[Zero Divide - TFK](https://remnant2.wiki.fextralife.com/Zero+Divide)

So I found this in N'Erud. Are there any specific restricted items? I have not seen any warnings or limitations on these items so far.

![2024-05-02_16-22-05](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/3e807619-cdea-44ca-af09-d141bdb39b94)

**Comment by razeedazee** [comment]
I assume the DLC exclusive items must be those locked behind DLC content, like storylines, bosses and stuff. The rest appear to be linked to a game update more than the dlc itself?

**Comment by razeedazee** [comment]
I think if i carried on rolling, I'd find any item that is listed as random area spawn on the wiki. The ones that are in specific locations   in certain areas, but random per roll, essentially found via exploration or content that was introduced via update which may have been bundled with the DLC. I am not aware of any new non random content though apart from things like Dwells new items.

The ones I won't see are will be in DLC area injectables/areas and the other stuff already mentioned.

AFK, the wiki stuff is datamined so maybe anything listed like

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/7b742cfc-59fd-4d10-9de0-1677600ebb3f)

**Comment by razeedazee** [comment]
Ring of Ordnance

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/abd0cd86-fc96-468c-98ec-3e583a2b7520)

Fextra is trash

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/c29b8f0a-1dd1-4fef-b2f3-9c9d267ee449)

**Comment by AndrewSav** [comment]
For the items that do not spawn randomly at least for DLC1 we have this bit of evidence:

<https://www.pcgamesn.com/remnant-2/dlc-play-without-buying#:~:text=%E2%80%9CJoin%20DLC%20owners'%20games%20in,up%20a%20copy%20for%20yourself.%E2%80%9D>

It says that you cannot "use any DLC-related items that you acquire" what ever it means. It implies thought that as far as RSG is concerned you still can get them.

> So would still a good idea to mark/filter DLC items in missing Item Tab. 

What would be the advantage of filtering them out in missing Item Tab if they can be acquired both with and without DLC?

>So people know what they can obtain with the DLCs they own.

I think the answer is the same regardless of the DLCs they own - random drops they will find in their own world, and story items  they can find in the multiplayer is this wrong? I mean I'm all for new features, just trying to understand when/how it's useful.

**Comment by AndrewSav** [comment]
Another datapoint: <https://steamcommunity.com/app/1282100/discussions/0/4039229229050419198/?ctp=2>

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/658865/34596848-848f-4b19-94ac-5c86306941ad)

**Comment by razeedazee** [comment]
From my point of view, many of these items are base game items, the ones I can collect with no warning. They were just bundled with DLC but are base game updates. I don't think they should be classified as DLC.

The items that are strictly limited to DLC with warnings, you won't see in your RSG anyway as you don't have the DLC. RGS is always relevant to my local experience. 

Maybe they can have some special attention but really you have two categories of items here, base game and unique DLC.

**Comment by AndrewSav** [comment]
> The items that are strictly limited to DLC with warnings, you won't see in your RSG anyway
I think that @TheNasbit point is that you will see them _on the missing items tab_

**Comment by razeedazee** [comment]
Yeah, i don't think it would be bad if there was some contextual awareness about which DLC those specific items belong to, the rest are just normal base game items.

I would find that helpful. Just not needed for the base game items.

**Comment by AndrewSav** [comment]
Now the problem with that is it's not clear which is which in an absence of a good data source. For example Ring of Ordnance looks like this in your save: `/Game/World_DLC2/Items/Trinkets/Rings/RingOfOrdnance/Ring_RingOfOrdnance.Ring_RingOfOrdnance_C` So it is associated with World_DLC2 and yet, you can (apparently) get it and use it without.

**Comment by razeedazee** [comment]
But i don't have the DLC so it's a base game item that came with an update.

Do the unique/restricted DLCS items have a special/different path?

**Comment by razeedazee** [comment]
would providing a profile/save game help btw?

**Comment by AndrewSav** [comment]
>But i don't have the DLC so it's a base game item that came with an update. 

I have a suspicion that the game files are (mostly) the same with and without DLCs, or you would not be able to join a multiplayer game with DLC so all resources are there. It's just a switch somewhere indicating if you made a purchase or not. Do in this sense it's a DLC item they decided that non-DLC owners has access to. And we do not seem to have a way to tell those from the others that cannot be accessed.

**Comment by razeedazee** [comment]
One limitation is the area/injectable? So far any item spawning in the base game world areas is not restricted. I figure these are the base game items?

**Comment by AndrewSav** [comment]
> One limitation is the area/injectable? 

Yes, but that only affects your ability to acquire it, not your ability to use it?

**Comment by AndrewSav** [comment]
Would be interesting if you could try and roll this Wood Ring from the screen shot @razeedazee , do you get the same entitlement tooltip still?

**Comment by razeedazee** [comment]
Ok, so in the remnant.wiki discord i asked as they had it listed as a bug, they tested it to see if it was fixed with new updates, it appears to be gone.

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/517941c5-ca56-47b6-8455-061cf5318c61)

I will have to go find it to test, but this was apparently the only base game area drop to do this.

**Comment by ThreePixiesInATrenchcoat** [comment]
Just disabled the DLC in steam to test, seeing DLC required on all items associated with specifically DLC areas, world drop items added in the same updates as DLC releases do not require DLC, Wood Ring was previously bugged to require TAK but I can confirm that's been fixed. remnant.wiki strives to have DLC required banners on the pages that do, I'll go over them today or tomorrow to add any DLC banners we may have forgotten.

**Comment by AndrewSav** [comment]
Also @razeedazee can you get <https://remnant2.wiki.fextralife.com/Steel+Scythe> from Brabus? It was added in DLC1. If I create a brand new char I can see it in Brabus shop immediately

**Comment by ThreePixiesInATrenchcoat** [comment]
![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/149244779/6eb7a5b2-7ac3-4807-8411-e0abb488ec34)
Steel Scythe doesn't require DLC, just added with it

**Comment by razeedazee** [comment]
@AndrewSav yeah I got that, the shovel, the new mods from Dwell.

**Comment by AndrewSav** [comment]
@ThreePixiesInATrenchcoat 

>Just disabled the DLC in steam to test

Gosh, I did not know that you can do that, it should simplify life a bit, thank you for sharing! 

> @AndrewSav yeah I got that, the shovel, the new mods from Dwell.

Thank you. Then I think you are right that we can filter based on if it's location related or not. Random drops / Vendors are for everyone and Locations, etc for DLC owners.

**Comment by razeedazee** [comment]
Just to confirm, i went and collected it.

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/0c7dbe22-89df-4ac2-a657-4ce48cce9f5d)

**Comment by ThreePixiesInATrenchcoat** [comment]
I went through every item from both DLC updates, they all fit the pattern of requiring the DLC if and only if they're tied to a DLC location (be it the story, a dungeon, or an injectable), except for these, which are all missing the tag, I've also posted about this in the official discord's bug report channel so the devs will hopefully be aware soon

Mutators:
Far-Sighted
Gladiator
Near-Sighted
Pressure Point
Searing Wounds
Thousand Cuts

Hats:
Dandy Topper
True Crimson Crown (crafted from the Crimson Shroud, which does require The Awakened King)
Mudtooth's Hat (not super sure if The Forgotten Kingdom is actually required for this one, requirements to get Mudtooth to tell this story are pretty opaque)

Consumables:
Egg Drink
Meatshake
Sweet Leaf (don't actually know if this is a dlc exclusive or just added to the pool you can get from chests in Yaesha)

**Comment by AndrewSav** [comment]
@ThreePixiesInATrenchcoat sorry can you clarify, what about those items that you listed? what does "missing the tag" mean?

**Comment by TheNasbit** [comment]
> Does RSG show consumables?
Not yet - but i though about adding which one are unlocked on the character in missing items.

**Comment by razeedazee** [comment]
@AndrewSav I interpret it to mean I could collect and use those items from a online world with the DLC even though I don't own the DLC.

Therefore, they are DLC unique items but not limited by the DLC only warning, which could be considered a bug/oversight but it might be intentional.

**Comment by AndrewSav** [comment]
@razeedazee I added the dlc filter to my fork as discussed above. And all the dlc 2 items. You can find it here: <https://github.com/AndrewSav/Remnant2SaveAnalyzer>. Let me know if you decided to give it a try if it worked for you.

**Comment by razeedazee** [comment]
@AndrewSav I will try it (in a bit) and get back to you.

**Comment by razeedazee** [comment]
I just did some Losomn, got a few random drops.  

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/4e3dbcd4-dcdc-4798-af70-4e88d4712d52)

These i can't get, except maybe the corrupted stuff if i farm.

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/141931230/fff8c8c9-ee16-4808-87b8-7647b15ea8ab)

I've exhausted Mudtooth's dialogue and no hat, If it's anything like before the unlock appears linked to some progression, like clearing specific biomes. I'd have to test if it's save or profile related, by trying his dialogue in another players world who base completed relevant DLC.

Feedback for App, i know you can look at missing items to see details, or click on item/wiki link but i think it would be useful is this context was merged in the app.

I'd prefer to click on the item and see the missing item entry, not go to the wiki.

**Comment by AndrewSav** [comment]
You should be able to mouse over an item to see the description:

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/658865/d40dc412-896f-45bf-b1e8-d1134469863a)

For "Can't get stuff"  did you disable Show DLC in settings?

![image](https://github.com/Razzmatazzz/RemnantSaveGuardian/assets/658865/7f5723fc-abc5-4c3b-94bc-35eb2ba2d6f3)

Corrupted items are not marked as DLC specific, I can certainly mark them if you know which is which. I do not check if you have corrupted shards or scrap to craft them of course, but otherwise. why cannot you get them?

**Comment by AndrewSav** [comment]
> You know what, i never knew that tooltip was there for the delay, i had always right clicked by then, lol. It's been there this whole time.

Same here, until someone else pointed that out to me

**Comment by AndrewSav** [comment]
Not everyone will like quicker, may be a setting for the tooltip delay

**Comment by razeedazee** [comment]
Personally, i think the tooltip is fine it's just context menu + tooltip interaction is slightly off. I think a delay slider is an elegant solution but it should default to a fast speed because tooltips are typically instant info delivered to the user.

**Comment by AndrewSav** [comment]
> Most DLC Items have a Tooltip in the main branch as well, at first I thought I was "overdoing" tooltips since items I added had way more tooltips than basegame Items. But I kinda like the extra information on the items, tho.

@TheNasbit  I do not mind if you use [mine](https://github.com/AndrewSav/lib.remnant2.analyzer/blob/main/db.json) as you see fit, it's open source after all
